src/rpc/core_rpc_server.cpp

```diff
@@ -205,14 +205,16 @@ namespace cryptonote
       }
       size_t txidx = 0;
       ntxes += bd.second.size();
-      for(const auto& t: bd.second)
+      for (std::list<cryptonote::blobdata>::iterator i = bd.second.begin(); i != bd.second.end(); ++i)
       {
+        unpruned_size += i->size();
         if (req.prune)
-          res.blocks.back().txs.push_back(get_pruned_tx_blob(t));
+          res.blocks.back().txs.push_back(get_pruned_tx_blob(std::move(*i)));
```

Comment:
This move won't happen because `get_pruned_tx_blob` takes the argument by const reference. And I do not think it easy to re-use the buffer currently - theres no harm in leaving it though. If the function is every updated to take by copy and reuse buffer this line will become more efficient.